src/modules/fancyzones/tests/UnitTests/Zone.Spec.cpp

```diff
@@ -5,48 +5,396 @@ using namespace Microsoft::VisualStudio::CppUnitTestFramework;
+        {
+            winrt::com_ptr<IZone> zone = MakeZone(m_zoneRect);
+
+            HWND window = Mocks::Window();
+            for (int i = 0; i < 10; i++)
+            {
+                HWND zoneWindow = Mocks::Window();
+                zone->AddWindowToZone(window, zoneWindow, i % 2 == 0);
+            }
+
+            Assert::IsFalse(zone->IsEmpty());
+        }
+
+        TEST_METHOD(IsNonEmptyMany)
+        {
+            winrt::com_ptr<IZone> zone = MakeZone(m_zoneRect);
+
+            for (int i = 0; i < 10; i++)
+            {
+                HWND zoneWindow = Mocks::Window();
+                HWND window = Mocks::Window();
+                zone->AddWindowToZone(window, zoneWindow, i % 2 == 0);
+            }
+
+            Assert::IsFalse(zone->IsEmpty());
+        }
+
+        TEST_METHOD(ContainsWindowEmpty)
+        {
+            winrt::com_ptr<IZone> zone = MakeZone(m_zoneRect);
+            HWND newWindow = Mocks::Window();
+            Assert::IsFalse(zone->ContainsWindow(newWindow));
+        }
+        TEST_METHOD(ContainsWindowNot)
+        {
+            winrt::com_ptr<IZone> zone = MakeZone(m_zoneRect);
+
+            for (int i = 0; i < 10; i++)
+            {
+                HWND zoneWindow = Mocks::Window();
+                HWND window = Mocks::Window();
+                zone->AddWindowToZone(window, zoneWindow, i % 2 == 0);
+            }
+
+            HWND newWindow = Mocks::Window();
+            Assert::IsFalse(zone->ContainsWindow(newWindow));
+        }
+
+        TEST_METHOD(ContainsWindowStampTrue)
+        {
+            winrt::com_ptr<IZone> zone = MakeZone(m_zoneRect);
+
+            HWND window = Mocks::Window();
+            HWND zoneWindow = Mocks::Window();
+            zone->AddWindowToZone(window, zoneWindow, true);
+
+            Assert::IsTrue(zone->ContainsWindow(window));
+        }
+
+        TEST_METHOD(ContainsWindowStampFalse)
+        {
+            winrt::com_ptr<IZone> zone = MakeZone(m_zoneRect);
+
+            HWND window = Mocks::Window();
+            HWND zoneWindow = Mocks::Window();
+            zone->AddWindowToZone(window, zoneWindow, false);
+
+            Assert::IsTrue(zone->ContainsWindow(window));
+        }
+
+        TEST_METHOD(ContainsWindowManyWindows)
+        {
+            winrt::com_ptr<IZone> zone = MakeZone(m_zoneRect);
+
+            HWND zoneWindow = Mocks::Window();
+            std::vector<HWND> windowVec{};
+            for (int i = 0; i < 10; i++)
+            {
+                HWND window = Mocks::Window();
+                windowVec.push_back(window);
+                zone->AddWindowToZone(window, zoneWindow, i % 2 == 0);
+            }
+
+            for (auto wnd : windowVec)
+            {
+                Assert::IsTrue(zone->ContainsWindow(wnd));
+            }
+        }
+
+        TEST_METHOD(ContainsWindowManyZoneWindows)
+        {
+            winrt::com_ptr<IZone> zone = MakeZone(m_zoneRect);
+
+            HWND window = Mocks::Window();

```

Comment:
Consider exporting following lines (adding many zone windows) to a function to avoid code duplication, there are few occurrences of this.